projects/igniteui-angular/src/lib/grids/columns/column-group.component.ts

```diff
@@ -243,7 +239,7 @@ export class IgxColumnGroupComponent extends IgxColumnComponent implements After
      * @memberof IgxColumnGroupComponent
      */
     get selected(): boolean {
-        return this.selectable && this.children.toArray().every(c => c.hidden || !c.selectable || c.selected);
+        return this.selectable && this.children.toArray().every(c => !c.selectable || c.selected);
```

Comment:
based on the getter above, **this.selectable** should return if any of it's child is selectable, which I think is not necessary to double-check in the callback here. Isn't it going to be easier if we can just try to find at least one child item who has been selected? Because it's really confusing at first glance. 
e.g
```typescript
    this.children.toArray().find(c => c.selected);
```

Reply:
to can check the selected property, we need to be sure that there at least one slectable child  and all the children, which are selectable are selected. Group will be selected only in case when it has selectable columns(this.selectable) and all the selectbale colums are selected. if you check only c.selected this will not work when there are selectable and not selectable columns in the group.